pocs/camera/pyro.py

```diff
@@ -0,0 +1,435 @@
+import sys
+import os
+import glob
+import time
+from warnings import warn
+from threading import Event
+from threading import Timer
+from threading import Thread
+from threading import Lock
+import subprocess
+
+from astropy import units as u
+import Pyro4
+
+from pocs.base import PanBase
+from pocs.utils import config
+from pocs.utils.logger import get_root_logger
+from pocs.utils import load_module
+from pocs.camera import AbstractCamera
+
+
+# Enable local display of remote tracebacks
+sys.excepthook = Pyro4.util.excepthook
+
+
+class Camera(AbstractCamera):
+    """
+    Class representing the client side interface to a distributed array of cameras
```

Comment:
Not a single camera in the array? I'm curious why that approach vs. exposing a set of remote cameras.

Reply:
It made sense at the time? I have, to begin with, been trying to avoid changes to `Observatory`, as part of my general approach of making these changes as self contained as possible and leaving all existing interfaces unchanged, & a single camera array object facilitated that.

Treating an array of cameras as if it were one camera is becoming increasingly awkward as I work my way up to the higher level functions, though, so it may be time to rethink that. Would need to modify `Observatory._create_cameras` and add the functions for finding & (re)connecting to remote cameras but it wouldn't be a major change, and would make implementing `take_observation` somewhat tidier.

Reply:
Thanks for raising this point. I've done a rewrite with single cameras and, at the cost of a small amount of additional code in `Observatory`, it's made things a _lot_ simpler. I'd previously been tying myself in knots trying to find a neat way to adapt `take_observation` to work with an array of cameras but with single cameras I didn't need to re-implement it at all, the base class version is fine.